src/bb/image-io/rt_u3v.h

```diff
@@ -322,443 +300,546 @@ class U3V {
+    {
+        if (instances_.count(id) == 0) {
+            ion::log::info("Create U3VRealCam instance: {}", id);
+            instances_[id] = std::unique_ptr<U3V>(new U3VRealCam(num_sensor, frame_sync, realtime_display_mode, sim_mode, width, height, fps, pixel_format));
+        }
+
+        return *instances_[id].get();
+    }
+
+
+    void set_realtime(std::vector<ArvBuffer *> &bufs, int timeout_us = 3 * 1000 * 1000){
+        std::vector<int32_t> N_output_buffers(num_sensor_);
+        for (auto i = 0; i < num_sensor_; ++i) {
+            int32_t num_input_buffer;
+            arv_stream_get_n_buffers(devices_[i].stream_, &num_input_buffer, &(N_output_buffers[i]));
+        }
+        // if all stream has N output buffers, discard N-1 of them
+        for (auto i = 0; i < num_sensor_; ++i) {
+            for (auto j = 0; j < N_output_buffers[i] - 1; ++j) {
+                bufs[i] = arv_stream_timeout_pop_buffer(devices_[i].stream_, timeout_us);
+                if (bufs[i] == nullptr) {
+                    log::error("pop_buffer(L2) failed due to timeout ({}s)", timeout_us * 1e-6f);
                     throw ::std::runtime_error("buffer is null");
                 }
                 devices_[i].frame_count_ = is_gendc_
-                    ? static_cast<uint32_t>(get_frame_count_from_genDC_descriptor(bufs[i], devices_[i]))
-                    : static_cast<uint32_t>(arv_buffer_get_timestamp(bufs[i]) & 0x00000000FFFFFFFF);
-
+                                           ? static_cast<uint32_t>(get_frame_count_from_genDC_descriptor(
+                                bufs[i], devices_[i]))
+                                           : static_cast<uint32_t>(arv_buffer_get_timestamp(bufs[i]) &
+                                                                   0x00000000FFFFFFFF);
                 i == 0 ?
-                    log::trace("All-Popped Frames (USB0, USB1)=({:20}, {:20})", devices_[i].frame_count_, "") :
-                    log::trace("All-Popped Frames (USB0, USB1)=({:20}, {:20})", "", devices_[i].frame_count_);
+                log::trace("All-Popped Frames (USB0, USB1)=({:20}, {:20}) [skipped for realtime display]",
+                           devices_[i].frame_count_, "") :
+                log::trace("All-Popped Frames (USB0, USB1)=({:20}, {:20}) [skipped for realtime display]",
+                           "", devices_[i].frame_count_);
+                arv_stream_push_buffer(devices_[i].stream_, bufs[i]);
+
             }
+        }
+    }
 
-            if (frame_sync_) {
-                uint32_t max_cnt = 0;
-                while (true) {
-                    // Update max_cnt
-                    for (int i=0; i<num_sensor_; ++i) {
-                        if (max_cnt < devices_[i].frame_count_) {
-                            max_cnt = devices_[i].frame_count_;
-                        }
+    void get(std::vector<Halide::Buffer<>>& outs) override{
+        auto timeout_us = 30 * 1000 * 1000;
+        if (sim_mode_){
```

Comment:
Is sim_mode in `U3VRealCam` required? It looks force_sim_mode leads to use `U3VFakeCam` class, not sim_mode in `U3vRealCam`.

Reply:
@Fixstars-iizuka if u3vRealCam doesn't detect any device it will fall back to sim_mode
for example:
```
        Node n = b.add("image_io_u3v_cameraN_u8x2")().set_param(
                Param("num_devices", num_device),
                Param("pixel_format", "Mono8")
);
```